cmd/root/root.go

```diff
@@ -151,5 +161,40 @@ Stack Trace:
 		}
 	}
 
+	exitCode := 0
+	if err != nil {
+		exitCode = 1
+	}
+
+	uploadTelemetry(cmd.Context(), commandString(cmd), startTime, exitCode)
 	return err
 }
+
+func uploadTelemetry(ctx context.Context, cmdStr string, startTime time.Time, exitCode int) {
+	// Return early if there are no logs to upload.
+	if !telemetry.HasLogs(ctx) {
+		log.Debugf(ctx, "no telemetry logs to upload")
+		return
+	}
+
+	// Telemetry is disabled. We don't upload logs.
+	if os.Getenv(telemetry.DisableEnvVar) != "" {
```

Comment:
Mix of `os.Getenv` and `env.Get`.

Reply:
fixed.

Reply:
not a suggestion, just a question, why os.Getenv isntead of env.Get given that we have context here?